#!/usr/bin/env python
PACKAGE = "nmpc_planner"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("neighborDistThreshold",  double_t, 0, "neighborDistThreshold", 3.0,  1, 20)

gen.add("approachAngleThreshold", double_t, 0, "approachAngleThreshold", 0.5, 0 , 20)

gen.add("distanceThresholdToTarget",  double_t, 0, "distanceThresholdToTarget", 8.0,  3, 15)

gen.add("copterDesiredHeightinNED",  double_t, 0, "copterDesiredHeightinNED", -8.0,  -20.0, -2.0)

gen.add("maxOffsetUncertaintyRadius", double_t, 0, "maxOffsetUncertaintyRadius", 3.0, 0 , 20)

gen.add("targetGuaranteeThreshold", double_t, 0, "targetGuaranteeThreshold", 7.0, 0 , 20)

gen.add("activeGuaranteeThreshold", double_t, 0, "activeGuaranteeThreshold", 1, 0 , 20)

gen.add("obstacleGuaranteeThreshold", double_t, 0, "obstacleGuaranteeThreshold", 1.5, 0 , 20)

gen.add("neighborGuaranteeThreshold", double_t, 0, "neighborGuaranteeThreshold", 1.5, 0 , 20)

gen.add("INTERNAL_SUB_STEP",  double_t, 0, "INTERNAL_SUB_STEP", 0.1,  0, 20)

gen.add("deltaT",  double_t, 0, "deltaT", 0.1,  0, 5.0)

gen.add("activeTrackingWeight",  double_t, 0, "activeTrackingWeight", 100,  0, 1000)
gen.add("energyWeight",  double_t, 0, "energyWeight", 10,  0, 1000)

gen.add("copterVelocityLimitHorizontal", double_t, 0, "copterVelocityLimitHorizontal", 5,  0, 10.0)
gen.add("copterVelocityLimitVertical", double_t, 0, "copterVelocityLimitVertical", 0.5,  0, 10.0)
gen.add("copterAccelarationLimitHorizontal", double_t, 0, "copterAccelarationLimitHorizontal", 5,  0, 20.0)
gen.add("copterAccelarationLimitVertical", double_t, 0, "copterAccelarationLimitVertical", 11,  0, 20.0)

exit(gen.generate(PACKAGE, "nmpc_planner", "nmpcPlannerParams"))
